--- a/src/main/formats/FFT/FFTInstr.cpp
+++ b/src/main/formats/FFT/FFTInstr.cpp
@@ -81,7 +81,7 @@
   //音色数だけ繰り返す。
   for (unsigned int i = 0; i <= hdr.iNumInstrs; i++) {
     //WdsInstr* newInstr = new WdsInstr(this,iOffset,sizeof(WdsRgnData),hdr.iBank,i);		//0 … hdr.iBank
-    WdsInstr *newInstr = new WdsInstr(this, iOffset, sizeof(WdsRgnData), i / 128, i % 128);        //0 … hdr.iBank
+    WdsInstr *newInstr = new WdsInstr(this, iOffset, sizeof(WdsRgnData), hdr.iBank, i);        // use WDS bank ID directly
     aInstrs.push_back(newInstr);
     //	newInstr->LoadInstr();		//VGMInstrSet::Load()関数内（LoadInstrs()）でやっている。
     iOffset += sizeof(WdsRgnData);    // size = 0x0010
